src/lib.rs

```diff
@@ -1179,7 +1187,7 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
     > {
         let user_key = match user_key {
             Some(user_key) => {
-                if self.sending_accounts.contains_key(&user_key.address()) {
+                if model.sending_accounts.get(&user_key.address()).is_ok() {
```

Comment:
I'm slightly concerned about this pattern, because _technically_ `Accounts::get` can fail if the account is not present _or_ if there is an error accessing storage. Not sure if this is really a problem.

Reply:
We could match on the type of error. Or have `load`/`get` return `Result<Option<T>, KeystoreError<L>>`. And then bubble the error up wherever we call `get`. Again, not sure this is a real problem/worthwhile. @bfish713 any thoughts?

Reply:
I think this one we should match on the error type and propagate up an error if it is not a `KeyNotFound` error.  It looks like here we only care if the Account exist so, another possibility is just add an `exists` function to `Accounts` which returns `Result<bool, KeystoreError<L>>` and does the same error handling.  I'm dealing with it for records as well where there are times we expect a record lookup to fail (like checking to see if a record is ours)

Reply:
I second `exists` for this use case. There are probably some other cases where treating `KeyNotFound` and `PersistenceError` the same is actually the right thing to do (e.g., during event handling when trying to find out if a record belongs to us, we look up the key, and whether the lookup fails or the key does not exist, all we can really do is forget that record). I think for all such cases, we should at least have a brief comment explaining why it's ok (if nothing else, it makes the tricky places easy to see and find)

Reply:
I just checked `Accounts::get` and  `KeyValueStore::load` again, and it looks like `KeyNotFound` is the only possible error. We are using the in-memory index, rather than the `AppendLog` store, when looking up the asset, so I don't think there will be a `PersistenceError`. There may be other places where we should handle different errors, but just not here.

Reply:
Oh I see we are keeping the whole storage in memory, I forgot about this detail.  I think this behavior might change when we have a true key value atomic storage though.

Reply:
Oh, good catch. And this probably won't (or doesn't need to) change even if/when we switch to using a real persistent KV-store, because there's no harm in still keeping an in-memory list of all the keys -- there should be relatively few of them and the public keys are not sensitive data. I guess this whole conversation is moot then. Hooray!